api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -352,6 +357,28 @@ type ControlPlaneConfigDump struct {
 	DumpSensitive ConfigDumpState `json:"dumpSensitive"`
 }
 
+// ControlPlaneObjectFilters defines filters to limit watched objects by the controllers.
+type ControlPlaneObjectFilters struct {
+	// Secrets defines the filters for watched secrets.
+	//
+	// +optional
+	Secrets *ControlPlaneFilterForSingleObject `json:"secrets,omitempty"`
+	// ConfigMaps defines the filters for watched config maps.
+	//
+	// +optional
+	ConfigMaps *ControlPlaneFilterForSingleObject `json:"configMaps,omitempty"`
+}
+
+// ControlPlaneFilterForSingleObject defines the filters for a certain type of object.
+type ControlPlaneFilterForSingleObject struct {
```

Comment:
I know naming is hard but I believe we could a bit better hear to get a clearer name.

WDYT about:

```suggestion
// ControlPlaneFilterForObjectType defines the filters for a certain type of object.
type ControlPlaneFilterForObjectType struct {
```

?

Reply:
Done.